pkgs/tools/networking/netbird/default.nix

```diff
@@ -0,0 +1,93 @@
+{ stdenv, lib, nixosTests, buildGoModule, fetchFromGitHub, installShellFiles
+, pkg-config
+, libayatana-appindicator, libX11, libXcursor, libXxf86vm
+, Cocoa, IOKit, Kernel, UserNotifications, WebKit
+, ui ? false }:
+let
+  modules = if ui then {
+    "client/ui" = "netbird-ui";
+  } else {
+    client = "netbird";
+    management = "netbird-mgmt";
+    signal = "netbird-signal";
+  };
+in
+buildGoModule rec {
+  pname = "netbird";
+  version = "0.8.6";
+
+  src = fetchFromGitHub {
+    owner = "netbirdio";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-POf/IEP1WRLJ5AMsvH76j97wJcQ2/dT91TyP0x/qnsg=";
+  };
+
+  vendorSha256 = "sha256-KtRQwrCBsOX7Jk9mKdDNOD7zfssADfBXCO1RPZbp5Aw=";
+
+  nativeBuildInputs = [ installShellFiles ] ++ lib.optional ui pkg-config;
+
+  buildInputs = lib.optionals (stdenv.isLinux && ui) [
+    libayatana-appindicator
+    libX11
+    libXcursor
+    libXxf86vm
+  ] ++ lib.optionals (stdenv.isDarwin && ui) [
+    Cocoa
+    IOKit
+    Kernel
+    UserNotifications
+    WebKit
+  ];
+
+  subPackages = lib.attrNames modules;
+
+  ldflags = [
+    "-s"
+    "-w"
+    "-X github.com/netbirdio/netbird/client/system.version=${version}"
+    "-X main.builtBy=nix"
+  ];
+
+  # needs network access
+  doCheck = false;
+
+  CGO_ENABLED = if ui then 1 else 0;
```

Comment:
```suggestion
  CGO_ENABLED = ui;
```

Reply:
`CGO_ENABLED` must be either 1 or 0, true or false would be wrong value.
https://pkg.go.dev/cmd/cgo
 > The cgo tool is enabled by default for native builds on systems where it is expected to work. It is disabled by default when cross-compiling. You can control this by setting the CGO_ENABLED environment variable when running the go tool: set it to 1 to enable the use of cgo, and to 0 to disable it.

Reply:
```
toString true == "1"
toString false = ""
```

If an empty string instead of 0 is ok, then indeed just doing `CGO_ENABLED = ui` should work.